elliptic-curve/src/hash2field/expand_msg_xmd.rs

```diff
@@ -1,72 +1,97 @@
-use super::ExpandMsg;
+use super::{Domain, ExpandMsg};
 use core::marker::PhantomData;
-use digest::{
+use digest_traits::{
     generic_array::{typenum::Unsigned, GenericArray},
     BlockInput, Digest,
 };
-use subtle::{Choice, ConditionallySelectable};
 
 /// Placeholder type for implementing expand_message_xmd based on a hash function
-#[derive(Debug)]
-pub struct ExpandMsgXmd<HashT> {
+pub struct ExpandMsgXmd<HashT>
+where
+    HashT: Digest + BlockInput,
+{
+    b_0: GenericArray<u8, HashT::OutputSize>,
+    b_vals: GenericArray<u8, HashT::OutputSize>,
+    domain: Domain<HashT::OutputSize>,
+    index: usize,
+    offset: usize,
+    ell: usize,
     phantom: PhantomData<HashT>,
```

Comment:
Now that `ExpandMsgXmd` holds data now, `PhantomData` isn't necessary anymore.

Reply:
Good call. I'll update.